CorsixTH/Lua/game_ui.lua

```diff
@@ -117,6 +117,24 @@ function GameUI:makeVisibleDiamond(scr_w, scr_h)
   }
 end
 
+--! Calculate the minimum valid zoom value
+--!
+--! Zooming out too much would cause negative width/height to be returned from
+--! makeVisibleDiamond. This function calculates the minimum zoom_factor that
+--! would be allowed.
+function GameUI:calculateMinimumZoom()
```

Comment:
world:adjustZoom has to deal with minimum zoom levels too, perhaps the logic can be consolidated.